types/src/nmt.rs

```diff
@@ -60,6 +63,12 @@ impl Namespace {
         Ok(Namespace(nmt_rs::NamespaceId(bytes)))
     }
 
+    pub fn new_v0_prefixed(id: [u8; NS_ID_V0_SIZE]) -> Self {
+        let mut bytes = [0u8; NS_SIZE];
+        bytes[NS_SIZE - NS_ID_V0_SIZE..].copy_from_slice(&id);
+        Namespace(nmt_rs::NamespaceId(bytes))
+    }
```

Comment:
this is not needed, `new_v0` should be able to handle it

Reply:
`new_v0` is fallible and makes extra checks redundant for this case, but I can unwrap instead in `is_tx()` and `is_pay_for_blob()`. I wanted `PAY_FOR_BLOB_NAMESPACE` and `TX_NAMESPACE` as constants, but it would require us to hardcode whole 29 bytes, what do you think?

Reply:
I see. I like one of these options:

1. Hardcode 29 bytes and give back a `Namespace` type as const
2. Same as 1 but introduce `const_namespace_v0` macro, which adds the 19 first bytes. When you define the const it will be like: `const_namespace_v0!(0, 0, 0, 0, 0, 0, 0, 0, 0, 1)`
3. Create a const fn `Namespace::const_v0`. The [`concat_bytes!`] would be very useful but it is unstable. Also for loop is not allowed in const fn, but.. we can have the following. I know it is a bit ugly but I will accept it for the sake of the end result.

```rust
pub const TX_NAMESPACE: Namespace = Namespace::const_v0([0, 0, 0, 0, 0, 0, 0, 0, 0, 1]);
pub const PAY_FOR_BLOB_NAMESPACE: Namespace = Namespace::const_v0([0, 0, 0, 0, 0, 0, 0, 0, 0, 4]);

// ....
impl Namespace {
    pub const fn const_v0(id: [u8; NS_ID_V0_SIZE]) -> Self {
        let mut bytes = [0u8; NS_SIZE];
        let start = NS_SIZE - NS_ID_V0_SIZE;

        bytes[start + 0] = id[0];
        bytes[start + 1] = id[1];
        bytes[start + 2] = id[2];
        bytes[start + 3] = id[3];
        bytes[start + 4] = id[4];
        bytes[start + 5] = id[5];
        bytes[start + 6] = id[6];
        bytes[start + 7] = id[7];
        bytes[start + 8] = id[8];
        bytes[start + 9] = id[9];

        Namespace(nmt_rs::NamespaceId(bytes))
    }
    
    pub fn is_tx(&self) -> bool {
        *self == TX_NAMESPACE
    }

    pub fn is_pay_for_blob(&self) -> bool {
        *self == PAY_FOR_BLOB_NAMESPACE
    }
 }
 ```
4. Check the needed slice:
```rust
    pub fn is_tx(&self) -> bool {
        self.version() == 0
            && self
                .as_bytes()
                .get(NS_SIZE - NS_ID_V0_SIZE..)
                .map_or(false, |b| b == &TX_NAMESPACE[..])
    }
```

[`concat_bytes!`]: https://doc.rust-lang.org/std/macro.concat_bytes.html

Reply:
![image](https://github.com/eigerco/celestia-node-rs/assets/34972409/5d0a7974-4ebd-4fdd-a0d1-934b0064711a)
cool, thanks!